### ISSUE TITLE
Are changes required for standalone operation with WireMock 2.14.0?

### ISSUE THREAD

**Issue by tillig** [MasonM/wiremock-jwt-extension#1]
I'm attempting to follow the instructions for standalone running using WireMock 2.14.0.

I have...

- Cloned the repo.
- Built the project (so I have `build\libs\wiremock-jwt-extension-0.1a.jar` and so on)
- Dropped the `wiremock-jwt-extension-0.1a.jar` into the same folder as `wiremock-standalone-2.14.0.jar` to make things easy.
- `java -cp wiremock-standalone-2.14.0.jar:wiremock-jwt-extension-0.1a.jar com.github.tomakehurst.wiremock.standalone.WireMockServerRunner --extensions="com.github.masonm.JwtMatcherExtension,com.github.masonm.JwtStubMappingTransformer"`

Unfortunately I get an exception:

```
Exception in thread "main" java.lang.NoClassDefFoundError: com/fasterxml/jackson/core/type/TypeReference
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Unknown Source)
        at com.github.tomakehurst.wiremock.extension.ExtensionLoader$3.apply(ExtensionLoader.java:75)
        at com.github.tomakehurst.wiremock.extension.ExtensionLoader$3.apply(ExtensionLoader.java:71)
        at wiremock.com.google.common.collect.Iterators$7.transform(Iterators.java:750)
        at wiremock.com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47)
        at wiremock.com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47)
        at wiremock.com.google.common.collect.Maps.uniqueIndex(Maps.java:1245)
        at wiremock.com.google.common.collect.Maps.uniqueIndex(Maps.java:1208)
        at com.github.tomakehurst.wiremock.extension.ExtensionLoader.asMap(ExtensionLoader.java:43)
        at com.github.tomakehurst.wiremock.extension.ExtensionLoader.loadExtension(ExtensionLoader.java:32)
        at com.github.tomakehurst.wiremock.extension.ExtensionLoader.load(ExtensionLoader.java:39)
        at com.github.tomakehurst.wiremock.standalone.CommandLineOptions.extensionsOfType(CommandLineOptions.java:303)
        at com.github.tomakehurst.wiremock.core.WireMockApp.<init>(WireMockApp.java:81)
        at com.github.tomakehurst.wiremock.WireMockServer.<init>(WireMockServer.java:70)
        at com.github.tomakehurst.wiremock.standalone.WireMockServerRunner.run(WireMockServerRunner.java:65)
        at com.github.tomakehurst.wiremock.standalone.WireMockServerRunner.main(WireMockServerRunner.java:113)
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.core.type.TypeReference
        at java.net.URLClassLoader.findClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        ... 17 more
```

Is there a change required to allow this to work with WireMock 2.14.0? Or am I doing something wrong?

Note I did try with the `wiremock-jwt-extension-0.1a-standalone.jar` and that seems to work, however it also appears to include all the wiremock server classes. I gather it is sufficient to just ensure the official WireMock standalone build is specified first on the classpath, but I figured I'd check to see if you're aware of any gotchas.

**Comment by tillig** [comment]
Possibly related: It appears if you run the extension standalone with the WireMock official standalone then the [call to `Json.getObjectMapper()` fails](https://github.com/MasonM/wiremock-jwt-extension/blob/8541b85b3787a58b815d288cbf8a995743403a82/src/main/java/com/github/masonm/Jwt.java#L41) silently and no JWT ever gets matched because parsing never completes.

**Comment by MasonM** [comment]
Sorry the delay, but I believe this was a ClassLoader issue and I was able to fix it by removing the dependency on `com.github.tomakehurst.wiremock.common.Json`. I also discovered a but where base64url values weren't decoded properly, resulting in false negatives, which I fixed by using `org.apache.commons.codec.binary.Base64` to decode, which properly handles base64url.

Version 0.3 has both fixes. Let me know if it works for you.
